test/ngMessages/messagesSpec.js

```diff
@@ -636,6 +636,27 @@ describe('ngMessages', function() {
     })
   );
 
+  it('should unregister the ngMessage even if it was never attached',
+    inject(function($compile, $rootScope) {
+      var html =
+        '<div ng-messages="items">' +
+          '<div ng-if="show"><div ng-message="x">ERROR</div></div>' +
+        '</div>';
+
+      element = $compile(html)($rootScope);
```

Comment:
There could be an expectation here that checks that the messge isn't initially attached / in the DOM

Reply:
Added